lib/util/stream_utils.js

```diff
@@ -407,13 +415,21 @@ shaka.util.StreamUtils = class {
    * @param {shaka.media.DrmEngine} drmEngine
    * @param {?shaka.extern.Variant} currentVariant
    * @param {shaka.extern.Manifest} manifest
+   * @param {?shaka.config.CodecSwitchingStrategy} codecSwitchingStrategy
    */
   static async filterManifest(
-      drmEngine, currentVariant, manifest) {
+      drmEngine, currentVariant, manifest, codecSwitchingStrategy) {
     await shaka.util.StreamUtils.filterManifestByMediaCapabilities(manifest,
         manifest.offlineSessionIds.length > 0);
-    shaka.util.StreamUtils.filterManifestByCurrentVariant(
-        currentVariant, manifest);
+
+    const Strategy = shaka.config.CodecSwitchingStrategy;
+    if (codecSwitchingStrategy && codecSwitchingStrategy ===
+       Strategy.DISABLED) {
+      shaka.util.StreamUtils.filterManifestByCurrentVariant(
```

Comment:
Do we really want to not filter the manifest at all if audio codec switching is enabled? This method filters for both video and audio, not just audio.

Reply:
Now with audio support, we feel this is the appropriate filtration. Please advise if still an issue.